chalk/core.py

```diff
@@ -189,35 +198,49 @@ def _repr_svg_(self) -> str:
         os.unlink(f.name)
         return svg
 
-    def atop(self, other: "Diagram") -> "Diagram":
+    def _empty_check(
+        self, other: Diagram
+    ) -> Tuple[Optional[Diagram], BoundingBox, BoundingBox]:
+        fake = BoundingBox([Point(0, 0)])
         box1 = self.get_bounding_box()
         box2 = other.get_bounding_box()
-        new_box = box1.union(box2)
+        if box2 is None:
+            return self, fake, fake
+        if box1 is None:
+            return other, fake, fake
+        return None, box1, box2
+
+    def atop(self, other: Diagram) -> Diagram:
+        dia, box1, box2 = self._empty_check(other)
+        if dia is not None:
+            return dia
+        new_box = BoundingBox.from_shapes([box1, box2])
         return Compose(new_box, self, other)
 
     __add__ = atop
 
-    def beside(self, other: "Diagram") -> "Diagram":
-        box1 = self.get_bounding_box()
-        box2 = other.get_bounding_box()
-        dx = box1.right - box2.left
-        t = tx.Translate(dx, 0)
-        new_box = box1.union(box2.apply_transform(t))
+    def _merge(self, other: Diagram, direction: Vec2) -> Diagram:
```

Comment:
A small observation. It might have been nice for this function to work on arbitrary `direction` vectors; for example, I would have expected the following snippet
```python
square(1)._merge(circle(0.5), 2 * unit_x + unit_y)
```
to place the bounding boxes of the two diagrams touching along the `2 * unit_x + unit_y` axis.

But again, I guess this functionality is already present in the envelopes branch.

Reply:
Yup that makes sense. I will define this by adding an optional argument to `atop`? 

`square(1).atop(circle(0.5),  2 * unit_x + unit_y)`

Thing that fits with the `diagrams` version.